docs/developer-documentation.md

```diff
@@ -1,32 +1,49 @@
 # Developer documentation
 
-The OpenEuropa Theme exposes ECL components as [UI Patterns][1] plugins, allowing them to be used seamlessly as drop-in
+The OpenEuropa Theme exposes ECL components using [UI Patterns][1] plugins, allowing them to be used seamlessly as drop-in
 templates for panels, field groups, views, paragraphs, nodes, etc.
 
 Patterns are located in [`./templates/patterns`](../templates/patterns), each sub-directory contains a pattern along with
-its variants. Pattern definitions are stored in a YAML file (`[PATTERN-NAME].ui_patterns.yml`) along with their Twig templates.
+its variants. Pattern definitions are stored in as YAML files (for ex. `[PATTERN-NAME].ui_patterns.yml`) along with their
+Twig templates (for ex. `pattern-[PATTERN-NAME].html.twig`).
 
-Each pattern definition exposes a list of fields that can be used to pass data to be rendered using the pattern template.
+Each pattern definition exposes a list of fields that can be used to pass data that will will be rendered using the
+related pattern template.
 
 Pattern fields can accept either one of the following types:
 
 - A renderable markup, i.e. a Drupal render array or a string
 - A value object or a list of value objects
 
+## Using value objects
+
 The scope of value objects is to make sure that data is passed to the final templates in a consistent and predictable way.
-Value objects are available at [`./src/ValueObject`](../src/ValueObject) and implement the `\Drupal\oe_theme\ValueObject\ValueObjectInterface`.
+Value objects are available at [`./src/ValueObject`](../src/ValueObject) and implement `\Drupal\oe_theme\ValueObject\ValueObjectInterface`.
 
-## Using value objects
+Value objects can be constructed only by using one of their factory methods. By default all value objects expose the
+`ValueObjectInterface::fromArray()` factory which builds and returns a value object from a given array.
 
-Value objects can be constructed only by using one of their factory methods. By default all value objects will expose the
-following factory:
+When patterns are rendered programmatically value objects can be passed directly to related pattern fields, as shown below:
 
-- `ValueObjectInterface::fromArray()`: build and return a value object from a given array.
+```php
+<?php
 
-For cases like component preview the Value objects must be constructed in pattern template preprocess functions, because 
-you can only pass the array from the YAML configuration.
+use Drupal\oe_theme\ValueObject\FooValueObject;
 
-Example how to use the value object:
+$elements['quote'] = [
+  '#type' => 'pattern',
+  '#id' => 'my_pattern',
+  '#fields' => [
+    'foo' => FooValueObject::fromArray(['bar' => 'Bar']),
+  ]
+];
+
+\Drupal::service('renderer')->render($elements);
+
+```
+
+Value objects will typically be constructed in pattern template preprocess functions, for example, when rendering a pattern
```

Comment:
Actually, they will typically be constructed in other preprocess functions, not those of patterns. Most notably the pattern preview because there we can only pass the values as array.

Reply:
That is not entirely true, core components will be using generic view mode preprocess function but site builders can get advantage of the modules that are shipped with UI Patterns, which can offer pattern theme suggestions based on rendering contexts, such as a view mode, a view row display etc.

Reply:
But I'll rephrase it as it is misleading the way it is now.